common/src/test/java/com/microsoft/identity/common/internal/ipc/IpcStrategyTests.java

```diff
@@ -0,0 +1,178 @@
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_GET_ACCOUNTS;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_GET_CURRENT_ACCOUNT_IN_SHARED_DEVICE;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_GET_DEVICE_MODE;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_GET_INTENT_FOR_INTERACTIVE_REQUEST;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_HELLO;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_REMOVE_ACCOUNT;
+import static com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation.MSAL_SIGN_OUT_FROM_SHARED_DEVICE;
+
+/**
+ * IMPORTANT: This class must cover EVERY {@link com.microsoft.identity.common.internal.broker.ipc.BrokerOperationBundle.Operation}.
+ */
+public abstract class IpcStrategyTests {
+
+    abstract IIpcStrategy getStrategy();
+
+    @Test
+    public void testMsalHello() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_HELLO));
+    }
+
+    @Test
+    public void testMsalGetIntentForInteractiveRequest() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_GET_INTENT_FOR_INTERACTIVE_REQUEST));
+    }
+
+    @Test
+    public void testMsalAcquireTokenSilent() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_ACQUIRE_TOKEN_SILENT));
+    }
+
+    @Test
+    public void testMsalGetAccounts() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_GET_ACCOUNTS));
+    }
+
+    @Test
+    public void testMsalRemoveAccounts() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_REMOVE_ACCOUNT));
+    }
+
+    @Test
+    public void testMsalGetDeviceMode() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_GET_DEVICE_MODE));
+    }
+
+    @Test
+    public void testGetCurrentAccountInSharedDevice() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_GET_CURRENT_ACCOUNT_IN_SHARED_DEVICE));
+    }
+
+    @Test
+    public void testMsalSignOutFromSharedDevice() {
+        testOperationSucceeds(getMockRequestBundle(MSAL_SIGN_OUT_FROM_SHARED_DEVICE));
+    }
+
+    @Test
+    public void testBrokerGetKeyFromInactiveBroker() {
+        testOperationNotSupportedOnClientSide(getMockRequestBundle(BROKER_GET_KEY_FROM_INACTIVE_BROKER));
+    }
+
+    @Test
+    public void testIpcFailed() {
+        testIpcConnectionFailed(getMockRequestBundle(MSAL_HELLO));
+    }
+
+    protected BrokerOperationBundle getMockRequestBundle(final BrokerOperationBundle.Operation operation) {
+        final Bundle bundle = new Bundle();
+        bundle.putBoolean("REQUEST_BUNDLE", true);
+
+        return new BrokerOperationBundle(operation, "MOCK_BROKER", bundle);
+    }
+
+    public static Bundle getMockIpcResultBundle() {
```

Comment:
Do we need to use a fake one?  Could we actually just pull a sample Bundle from the actual code?

Reply:
I intentionally did not use the actual sample bundle - since that depends on the protocol.
This test case should be protocol-agnostic.